extensions/grpc/src/main/java/com/hazelcast/jet/grpc/impl/UnaryService.java

```diff
@@ -49,14 +49,15 @@ public void destroy() {
     private static class Observer<R> implements StreamObserver<R> {
         private final CompletableFuture<R> future;
 
-        private volatile R value ;
+        private R value ;
 
         Observer() {
             this.future = new CompletableFuture<>();
         }
 
         @Override
         public void onNext(R value) {
+            assert this.value == null : "value should not be assigned twice in unary mode";
```

Comment:
This would be user error, so let's raise this to a hard check.

Reply:
grpc doesn't allow this, and throws exception. The only case would be if this observer was used for a non-unary service, whcih I think the signature wouldn't allow.